src/main/java/stream/flarebot/flarebot/util/general/GuildUtils.java

```diff
@@ -0,0 +1,311 @@
+     * @return null if the role doesn't, otherwise a list of roles matching the string
+     */
+    public static Role getRole(String s, String guildId) {
+        return getRole(s, guildId, null);
+    }
+
+    /**
+     * Gets a {@link Role} that matches a string. Case doesn't matter.
+     *
+     * @param s       The String to get a role from
+     * @param guildId The id of the {@link Guild} to get the role from
+     * @param channel The channel to send an error message to if anything goes wrong.
+     * @return null if the role doesn't, otherwise a list of roles matching the string
+     */
+    public static Role getRole(String s, String guildId, TextChannel channel) {
+        Guild guild = Getters.getGuildById(guildId);
+        Role role = guild.getRoles().stream()
+                .filter(r -> r.getName().equalsIgnoreCase(s))
+                .findFirst().orElse(null);
+        if (role != null) return role;
+        try {
+            role = guild.getRoleById(Long.parseLong(s.replaceAll("[^0-9]", "")));
+            if (role != null) return role;
+        } catch (NumberFormatException | NullPointerException ignored) {
+        }
+        if (channel != null) {
+            if (guild.getRolesByName(s, true).isEmpty()) {
+                String closest = null;
+                int distance = LEVENSHTEIN_DISTANCE;
+                for (Role role1 : guild.getRoles().stream().filter(role1 -> FlareBotManager.instance().getGuild(guildId).getSelfAssignRoles()
+                        .contains(role1.getId())).collect(Collectors.toList())) {
+                    int currentDistance = StringUtils.getLevenshteinDistance(role1.getName(), s);
+                    if (currentDistance < distance) {
+                        distance = currentDistance;
+                        closest = role1.getName();
+                    }
+                }
+                MessageUtils.sendErrorMessage("That role does not exist! "
+                        + (closest != null ? "Maybe you mean `" + closest + "`" : ""), channel);
+                return null;
+            } else {
+                return guild.getRolesByName(s, true).get(0);
+            }
+        }
+        return null;
+    }
+    //User
+
+    /**
+     * Gets a {@link User} from a string. Not case sensitive.
+     * The string can eater be their name, their id, or them being mentioned.
+     *
+     * @param s the string to get the user from
+     * @return null if the user wasn't found otherwise a {@link User}
+     */
+    public static User getUser(String s) {
+        return getUser(s, null);
+    }
+
+    /**
+     * Gets a {@link User} from a string. Not case sensitive.
+     * The string can eater be their name, their id, or them being mentioned.
+     *
+     * @param s       The string to get the user from.
+     * @param guildId The string id of the {@link Guild} to get the user from.
+     * @return null if the user wasn't found otherwise a {@link User}.
```

Comment:
The alignment here is a clusterfuck, align pls  ❤️

Reply:
JDA does that when you format code.

Reply:
Know any way to turn that off?

Reply:
Not off the top of my head, just put a few spaces there